curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md

```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
 
 #### --distractors--
 
-Placeholder distractor 1
+`package.json`
 
 ---
 
-Placeholder distractor 2
+`package.json`
 
 ---
 
-Placeholder distractor 3
+`typescript.config.js`
 
 #### --answer--
 
-Placeholder answer
+`tsconfig.json`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What does the `target` option in the `tsconfig.json` file specify?
 
 #### --distractors--
 
-Placeholder distractor 1
+The version of TypeScript to use during compilation.
 
 ---
 
-Placeholder distractor 2
+The target runtime environment (e.g., Node.js or browser) for the TypeScript code.
 
 ---
 
-Placeholder distractor 3
+The specific module system to use for imports and exports.
 
 #### --answer--
 
-Placeholder answer
+
+The ECMAScript version to which the TypeScript compiler will transpile the code.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What does the `noImplicitReturns` option in the `tsconfig.json` file enforce?
 
 #### --distractors--
 
-Placeholder distractor 1
+It forces all functions to return a `number`.
 
 ---
 
-Placeholder distractor 2
+It requires that all functions return either `true` or `false`.
 
 ---
 
-Placeholder distractor 3
+It prevents using the `any` type in TypeScript code.
 
 #### --answer--
 
-Placeholder answer
+It warns if a function does not return a value.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which TypeScript option prevents implicit `any` types?
 
 #### --distractors--
 
-Placeholder distractor 1
+`noImplicitReturns`
 
 ---
 
-Placeholder distractor 2
+`alwaysStrict`
 
 ---
 
-Placeholder distractor 3
+`strictNullChecks`
 
 #### --answer--
 
-Placeholder answer
+`noImplicitAny`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the purpose of the `never` type in TypeScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+To represent null or undefined.
 
 ---
 
-Placeholder distractor 2
+To represent any possible value.
 
 ---
 
-Placeholder distractor 3
+To represent a type that can be anything except null.
 
 #### --answer--
 
-Placeholder answer
+To represent the type of values that never occur.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What will be the output of the following TypeScript code?
+
+    enum Color {
+    	 Red,
+    	 Blue
+    }
+    console.log(Color.Red, Color.Blue);
+
 
 #### --distractors--
 
-Placeholder distractor 1
+`Red Blue`
 
 ---
 
-Placeholder distractor 2
+`1, 2`
 
 ---
 
-Placeholder distractor 3
+`Red 2`
 
 #### --answer--
 
-Placeholder answer
+`0 1`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+How can you specify that a function parameter is optional in TypeScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+By using the `optional` keyword
 
 ---
 
-Placeholder distractor 2
+By setting a default value
 
 ---
 
-Placeholder distractor 3
+By using the `any` type
 
 #### --answer--
 
-Placeholder answer
+By adding a `?` after the parameter name
+
+
 
```

Comment:
```suggestion
By adding a `?` after the parameter name

```

Reply:
Got it! I've made the suggested changes. Thanks